File: frontend/src/types/scenario.ts
/**
 * Matches the backend ScenarioResponse Pydantic schema exactly.
 * Backend fields: id (int), name, description,
 * created_at, updated_at (snake_case datetimes).
 *
 * Note: Backend uses 'name' — frontend components should use scenario.name.
 */
export interface Scenario {
  id: number;
  name: string;
  description?: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateScenarioDTO {
  name: string;
  description?: string;
}

export interface UpdateScenarioDTO extends Partial<CreateScenarioDTO> {}

File: backend/app/schemas/scenario.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class ScenarioBase(BaseModel):
    name: str
    description: str | None = None


class ScenarioCreate(ScenarioBase):
    pass


class ScenarioResponse(ScenarioBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime
    updated_at: datetime
